lib/transformer.dart

```diff
@@ -98,64 +98,73 @@ class WebSkinDartTransformer extends Transformer implements LazyTransformer {
     TransformedSourceFile transformedFile = new TransformedSourceFile(sourceFile);
     TransformLogger logger = new JetBrainsFriendlyLogger(transform.logger);
 
-    var partPattern = new RegExp(r'''part\s+['"](.+\.over_react\.g\.dart)['"];''');
-    var partFilename = partPattern.firstMatch(sourceFile.getText(0));
-
-    // For Dart 1 compatibility an empty generated part file will be created when a file contains
-    // the part directive pointing to the generated file the new builder requires.
-    if (partFilename != null) {
-      var sourceFileDirectory = p.dirname(transform.primaryInput.id.path);
-      var sourceFilename = p.basename(transform.primaryInput.id.path);
-      var partFilePath = p.join(sourceFileDirectory, partFilename.group(1));
-      var contents = "part of '$sourceFilename';";
-      var asset = new Asset.fromString(new AssetId(transform.primaryInput.id.package, partFilePath), contents);
-
-      transform.addOutput(asset);
-    }
+    if (transform.primaryInput.id.path.endsWith('.over_react.g.dart')) {
+      var partOfPattern = new RegExp(r'''part[ ]+of[ ]+['"].+\.dart['"];''');
+      var partOfMatch = partOfPattern.firstMatch(sourceFile.getText(0));
+      if (partOfMatch != null) {
+        transformedFile.remove(sourceFile.span(partOfMatch.end));
+      }
+    } else {
+      var partPattern = new RegExp(r'''part\s+['"](.+\.over_react\.g\.dart)['"];''');
+      var partFilename = partPattern.firstMatch(sourceFile.getText(0));
+
+      // For Dart 1 compatibility an empty generated part file will be created when a file contains
+      // the part directive pointing to the generated file the new builder requires.
+      if (partFilename != null) {
+        var sourceFileDirectory = p.dirname(transform.primaryInput.id.path);
+        var partFilePath = p.join(sourceFileDirectory, partFilename.group(1));
+        var partAssetId = new AssetId(transform.primaryInput.id.package, partFilePath);
+        if (!(await transform.hasInput(partAssetId))) {
```

Comment:
This is how we prevent a duplicate asset output transformer error when dealing with builder outputs being committed to source.